cli/waiter/cli.py

```diff
@@ -0,0 +1,93 @@
+                    version=f'%(prog)s version {version.VERSION}', action='version')
+
+subparsers = parser.add_subparsers(dest='action')
+
+actions = {
+    'create': create.register(subparsers.add_parser),
+    'show': show.register(subparsers.add_parser)
+}
+
+
+def load_target_clusters(config_map, url=None, cluster=None):
+    """Given the config and (optional) url and cluster flags, returns the list of clusters to target"""
+    if cluster and url:
+        raise Exception('You cannot specify both a cluster name and a cluster url at the same time')
+
+    clusters = None
+    config_clusters = config_map.get('clusters')
+    if url:
+        if urlparse(url).scheme == '':
+            url = 'http://%s' % url
+        clusters = [{'name': url, 'url': url}]
+    elif config_clusters:
+        if cluster:
+            clusters = [c for c in config_clusters if c.get('name').lower() == cluster.lower()]
+            if len(clusters) == 0 and len(config_clusters) > 0:
+                config_cluster_names = ', '.join([c.get('name') for c in config_clusters])
+                raise Exception(f'You specified cluster {cluster}, which was not present in your config.' +
+                                f' You have the following clusters configured: {config_cluster_names}.')
+        else:
+            clusters = [c for c in config_clusters if 'disabled' not in c or not c['disabled']]
+
+    return clusters
+
+
+def run(args):
+    """
+    Main entrypoint to the Waiter CLI. Loads configuration files,
+    processes global command line arguments, and calls other command line 
+    sub-commands (actions) if necessary.
+    """
+    args, unknown_args = parser.parse_known_args(args)
+    if args.action == 'create':
+        create.add_implicit_arguments(unknown_args)
+    args = parser.parse_args()
+    args = vars(args)
+
+    verbose = args.pop('verbose')
+
+    log_format = '%(asctime)s [%(levelname)s] [%(name)s] %(message)s'
+    if verbose:
+        logging.getLogger('').handlers = []
+        logging.basicConfig(format=log_format, level=logging.DEBUG)
+    else:
+        logging.disable(logging.FATAL)
+
+    logging.debug('args: %s' % args)
+
+    action = args.pop('action')
+    config_path = args.pop('config')
+    cluster = args.pop('cluster')
+    url = args.pop('url')
+
+    if action is None:
+        parser.print_help()
+    else:
+        config_map = configuration.load_config_with_defaults(config_path)
+        try:
+            metrics.initialize(config_map)
+            metrics.inc('command.%s.runs' % action)
+            clusters = load_target_clusters(config_map, url, cluster)
+            http.configure(config_map)
+            args = {k: v for k, v in args.items() if v is not None}
+            result = actions[action](clusters, args, config_path)
+            logging.debug('result: %s' % result)
+            return result
+        finally:
+            metrics.close()
```

Comment:
Should we include an execution time metric?

Reply:
I don't see this being so useful, but if you feel strongly, we can.

Reply:
Okay, we can skip for now.